pkg/clusterlogging/clusterlogforwarder.go

```diff
@@ -114,36 +114,42 @@ func (builder *ClusterLogForwarderBuilder) WithPipeline(pipelineSpec *clov1.Pipe
 }
 
 // PullClusterLogForwarder retrieves an existing clusterlogforwarder object from the cluster.
-func PullClusterLogForwarder(apiClient *clients.Settings, name, namespace string) (*clov1.ClusterLogForwarder, error) {
-	glog.V(100).Infof("Pulling existing clusterlogforwarder %s in namespace %s", name, namespace)
+func PullClusterLogForwarder(apiClient *clients.Settings, name, nsname string) (*ClusterLogForwarderBuilder, error) {
+	if apiClient == nil {
```

Comment:
Please move this verification after log message:
glog.V(100).Infof("Pulling existing clusterlogforwarder %s in nsname %s", name, nsname)